### PR TITLE
Properly support protocols between 740 and 770

### PR DESCRIPTION
**Pull request by Sifl46**
I noticed that the source supposedly had support for the 7.4 protocol. However, upon trying it out, I learned that it very much did not work correctly. I have taken it upon myself to fix not only 7.4 support, but all clients between 7.4 and 7.7.

### CODE REVIEW

File: src/client/protocolgameparse.cpp [hunk 1]
```diff
@@ -1587,7 +1595,10 @@ void ProtocolGame::parsePlayerStats(const InputMessagePtr& msg)
         m_localPlayer->setBaseMagicLevel(baseMagicLevel);
     }
 
-    const double soul = msg->getU8();
+    double soul = 0;
+    if (g_game.getFeature(Otc::GameSoulStat))
```
Comment (nekiro) [inline src/client/protocolgameparse.cpp]: rename to GameSoul

File: src/client/protocolgamesend.cpp [hunk 2]
```diff
@@ -70,19 +70,20 @@ void ProtocolGame::sendLoginPacket(uint challengeTimestamp, uint8 challengeRando
         msg->addU8(0);
 
     const int offset = msg->getMessageSize();
-    // first RSA byte must be 0
-    msg->addU8(0);
 
     if (g_game.getFeature(Otc::GameLoginPacketEncryption)) {
+        // first RSA byte must be 0
+        msg->addU8(0);
```
Comment (nekiro) [inline src/client/protocolgamesend.cpp:76]: arent you breaking other protocols by this change?

Comment (Sifl46) [reply src/client/protocolgamesend.cpp:76]: with the two bytes i moved around, it works out to be the exact same end result.
The change is simply for a more accurate code flow.
With the current code, the client "expects" the RSA header (which is hardcoded to be read as 0) whether encryption is enabled or not. While the gm byte is inside the encryption feature check (again hardcoded to read 0), while earlier protocols still look for it despite not supporting encryption.

I had also tested this change with 8.00 protocol, and it worked fine.

File: src/client/protocolgameparse.cpp (hunk 1 above)
Comment (Sifl46) [reply src/client/protocolgameparse.cpp]: Done.

File: src/client/protocolgamesend.cpp [hunk 3]
```diff
@@ -70,19 +70,20 @@ void ProtocolGame::sendLoginPacket(uint challengeTimestamp, uint8 challengeRando
...
         // xtea key
         generateXteaKey();
         msg->addU32(m_xteaKey[0]);
         msg->addU32(m_xteaKey[1]);
         msg->addU32(m_xteaKey[2]);
         msg->addU32(m_xteaKey[3]);
-        msg->addU8(0); // is gm set?
     }
 
+    msg->addU8(0); // is gm set?
```
Comment (Sifl46) [inline src/client/protocolgamesend.cpp:85]: See: here

File: src/client/protocolgameparse.cpp (hunk 1 above)
Comment (oarsc) [reply src/client/protocolgameparse.cpp]: This exact line @nekiro mentioned is not corrected in the commit 68d3189943be6da18bef730a4898acbb040d7cc4

Comment (Sifl46) [reply src/client/protocolgameparse.cpp]: Awkward. Commited a fix (and actually compiled it this time)

**Pull request merged**